### DISCUSSION TITLE
Javascript Obfuscation

### DISCUSSION THREAD

**Discussion by leandroribeir0** [crxjs/chrome-extension-tools#510] [Q&A]
Guys, someone can help me to setup an automatic JS Obfuscation on build?

I'm developing a extension with authentication and I would like to difficulty the readable of js code.

**Comment by palamarchukser** [comment]
you can choose the appropriate package in the list of obfuscate packages
https://npm.io/search/keyword:javascript+obfuscator

just  FYI:
```
Google disallows obfuscated code.
The specific policy, available at https://developer.chrome.com/webstore/program_policies
```
![image](https://user-images.githubusercontent.com/10272176/188587450-c6c1ecca-082f-433e-ae4f-cbc9ba545573.png)

**Reply by jacksteamdev** [reply]
Obfuscation in a Chrome extension will get your extension rejected by the CWS. The Chrome Web Store reviews your extension and needs to be able to read the code. 

Minification is OK. https://vitejs.dev/config/build-options.html#build-minify
